### PR TITLE
Return false if ini-file could not be opened.

### PR DESCRIPTION
**Pull request by invy**

### CODE REVIEW

File: src/CommonAPI/IniFileReader.cpp [hunk 1]
```diff
@@ -29,65 +29,66 @@ IniFileReader::Section::getValue(const std::string &_key) const {
...
+                currentSectionName = line.substr(++start, --end);
+                if (sections_.end() == sections_.find(currentSectionName)) {
+                    currentSection = std::make_shared<Section>();
+                    if (currentSection) {
+                        sections_[currentSectionName] = currentSection;
                     }
                 } else {
-                    COMMONAPI_ERROR("Missing \']\' in section definition (line ",
-                                    lineCounter, ")");
+                    COMMONAPI_ERROR("Double definition of section \'",
```
Comment (gunnarx) [inline src/CommonAPI/IniFileReader.cpp]: Seems like there should be more consideration regarding the return value.  What's the appropriate return value for these other COMMONAPI_ERROR cases?

File: src/CommonAPI/IniFileReader.cpp [hunk 2]
```diff
@@ -29,65 +29,66 @@ IniFileReader::Section::getValue(const std::string &_key) const {
...
-                    } else {
-                        std::string value = line.substr(pos+1);
-                        trim(value);
-                        currentSection->mappings_[key] = value;
-                    }
-                } else if (line.size() > 0) {
-                    COMMONAPI_ERROR("Missing \'=\' in key=value definition (line ",
-                                    lineCounter, ")");
+            } else {
+                COMMONAPI_ERROR("Missing \']\' in section definition (line ",
```
Comment (gunnarx) [inline src/CommonAPI/IniFileReader.cpp]: like here...

File: src/CommonAPI/IniFileReader.cpp [hunk 3]
```diff
@@ -29,65 +29,66 @@ IniFileReader::Section::getValue(const std::string &_key) const {
...
+                                lineCounter, ")");
+            }
+        } else if (currentSection) {
+            std::size_t pos = line.find('=');
+            if (pos != line.npos) {
+                std::string key = line.substr(0, pos);
+                trim(key);
+                if (currentSection->mappings_.end()
+                    != currentSection->mappings_.find(key)) {
+                    COMMONAPI_ERROR("Double definition for key \'",
```
Comment (gunnarx) [inline src/CommonAPI/IniFileReader.cpp]: ...and here...

File: src/CommonAPI/IniFileReader.cpp [hunk 4]
```diff
@@ -29,65 +29,66 @@ IniFileReader::Section::getValue(const std::string &_key) const {
...
+                                    "\' (line ",
+                                    lineCounter,
+                                    ")");
+                } else {
+                    std::string value = line.substr(pos+1);
+                    trim(value);
+                    currentSection->mappings_[key] = value;
                 }
+            } else if (line.size() > 0) {
+                COMMONAPI_ERROR("Missing \'=\' in key=value definition (line ",
```
Comment (gunnarx) [inline src/CommonAPI/IniFileReader.cpp]: ...

**Comment by GenivivSOMEIPmaintainer** [comment]
Has been integrated in CommonAPI 3.1.12.

**Pull request closed without merging**